### PR TITLE
[FLINK-25927][connectors] Make flink-connector-base dependency usage consistent across all connectors 

### PR DESCRIPTION
**Pull request by afedulov**
## What is the purpose of the change

`flink-connector-base` was previously inconsistently used in connectors (directly shaded in some and transitively pulled in via `flink-connector-files` which was itself shaded in the table uber jar). FLINK-24687 moved `flink-connector-files` out from the `flink-table` uber jar. This commit implements a combined approach for ensuring a smooth transition for both Flink users and for external connector developers, as outlined in this :
- all internal Flink connectors that depend on `flink-connector-base` now shade and relocate it
- for compatibility, until external developers implement the same change, `flink-connector-base` is also included into `flink-dist`

The rationale behind this combined approach was discussed [here](https://lists.apache.org/thread/hhjoc9hobs6h5pydr0623no2l36r6n6r).

## Brief change log
  - `flink-connector-base` dependency is shaded and relocated in all connectors
  - `flink-connector-base` is added into `flink-dist`

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no )
  - The serializer: ( no )
  - The runtime per-record code paths (performance sensitive): ( no )
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no )
  - The S3 file system connector: ( no )

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (release notes). The following text was added to the tickets' release notes:
  
## Known issues:
- Due to the setup of the newly added Kafka end-to-end tests, they became incompatible with this change when executed in the IDE (unshaded classes from IntelliJ modules get used in the `RemoteStreamEnvironment`). The testing approach is due to be revisited in the next versions of Flink.
- I had to modify an existing ArchUnit violation in order to get green CI because the rule correctly does not take into account if the caller of a @VisibleForTesting method is itself marked with the same annotation. A proper fix is here: https://issues.apache.org/jira/browse/FLINK-26186

### CODE REVIEW

**Comment by dmvk** [comment]
I did a quick sanity check for all connector artifacts:

```bash
find flink-connectors -type f -name '*.jar' | grep -v original | grep -v '\-test' | for i in `cat -`; do echo "==== $i ===="; jar tf $i | grep 'org/apache/flink/connector/base/source/reader/RecordEmitter'; done;
==== flink-connectors/flink-hcatalog/target/flink-hcatalog_2.12-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-elasticsearch7/target/flink-connector-elasticsearch7-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/elasticsearch7/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-hive-1.2.2/target/flink-sql-connector-hive-1.2.2_2.12-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-elasticsearch6/target/flink-connector-elasticsearch6-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/elasticsearch6/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-pulsar/target/flink-sql-connector-pulsar-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/pulsar/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-base/target/flink-connector-base-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-elasticsearch6/target/flink-sql-connector-elasticsearch6-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/elasticsearch6/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-kafka/target/flink-sql-connector-kafka-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kafka/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/kafka/sql/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-elasticsearch7/target/flink-sql-connector-elasticsearch7-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/elasticsearch7/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-kinesis/target/flink-sql-connector-kinesis-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-gcp-pubsub/target/flink-connector-gcp-pubsub-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-hive-2.3.6/target/flink-sql-connector-hive-2.3.6_2.12-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-jdbc/target/flink-connector-jdbc-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-hbase-base/target/flink-connector-hbase-base-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-hive-3.1.2/target/flink-sql-connector-hive-3.1.2_2.12-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-cassandra/target/flink-connector-cassandra_2.12-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-hive/target/flink-connector-hive_2.12-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-aws-kinesis-firehose/target/flink-connector-aws-kinesis-firehose-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/firehose/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-elasticsearch-base/target/flink-connector-elasticsearch-base-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-aws-base/target/flink-connector-aws-base-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-hive-2.2.0/target/flink-sql-connector-hive-2.2.0_2.12-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-kinesis/target/flink-connector-kinesis-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-file-sink-common/target/flink-file-sink-common-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-hbase-1.4/target/flink-sql-connector-hbase-1.4-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-kafka/target/flink-connector-kafka-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kafka/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-twitter/target/flink-connector-twitter-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-hadoop-compatibility/target/flink-hadoop-compatibility_2.12-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-files/target/flink-connector-files-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/files/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-aws-kinesis-data-streams/target/flink-connector-aws-kinesis-data-streams-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-elasticsearch5/target/flink-connector-elasticsearch5-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-hbase-1.4/target/flink-connector-hbase-1.4-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-rabbitmq/target/flink-sql-connector-rabbitmq-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-aws-kinesis-firehose/target/flink-sql-connector-aws-kinesis-firehose-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/firehose/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-sql-connector-aws-kinesis-data-streams/target/flink-sql-connector-aws-kinesis-data-streams-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-pulsar/target/flink-connector-pulsar-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/pulsar/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
==== flink-connectors/flink-connector-rabbitmq/target/flink-connector-rabbitmq-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-sql-connector-hbase-2.2/target/flink-sql-connector-hbase-2.2-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-nifi/target/flink-connector-nifi-1.15-SNAPSHOT.jar ====
==== flink-connectors/flink-connector-hbase-2.2/target/flink-connector-hbase-2.2-1.15-SNAPSHOT.jar ====
```

There seems to be some weird ones, eg.:

```bash
==== flink-connectors/flink-sql-connector-kinesis/target/flink-sql-connector-kinesis-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/kinesis/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
```

```
==== flink-connectors/flink-sql-connector-elasticsearch7/target/flink-sql-connector-elasticsearch7-1.15-SNAPSHOT.jar ====
org/apache/flink/connector/elasticsearch7/shaded/org/apache/flink/connector/base/source/reader/RecordEmitter.class
org/apache/flink/connector/base/source/reader/RecordEmitter.class
```

If I understand the change correctly, we should always have at most one `connector-base` per "published" connector artifact, is that correct? Also should it be always properly shaded?

File: flink-connectors/flink-connector-aws-kinesis-firehose/pom.xml [hunk 1]
```diff
@@ -164,6 +158,33 @@ under the License.
...
+							<artifactSet>
+								<includes>
+									<include>org.apache.flink:flink-connector-base</include>
+									<include>org.apache.flink:flink-connector-aws-base</include>
+								</includes>
+							</artifactSet>
+							<relocations>
+								<relocation>
+									<pattern>org.apache.flink.connector.base</pattern>
+									<shadedPattern>org.apache.flink.connector.firehose.shaded.org.apache.flink.connector.base</shadedPattern>
```
Comment (dmvk) [inline flink-connectors/flink-connector-aws-kinesis-firehose/pom.xml:181]:
```suggestion
									<shadedPattern>${shade.pattern}.org.apache.flink.connector.base</shadedPattern>
```
It would be nice to have shade pattern as a variable, to make sure it's aligned if we're shading more  than one dependency.

**Review by dmvk — COMMENTED** [review]
Thanks for the PR @afedulov! The overall sentiment of the PR looks really good  👍 This is headed in a right direction. Please try to go over the final artifacts (I've pasted an example above), there still seem to be some artifacts violating the new "contract".

I'm also wondering whether we'd be able to add some test (bash based?) for the shaded artifacts, making sure this contract will hold with future changes 🤔

**Comment by afedulov** [comment]
Thanks for the review, @dmvk, and for spotting this:
> I did a quick sanity check for all connector artifacts:
>...There seems to be some weird ones, eg.:

These were pulled in from the connector's transitive dependencies and packaged because of the "double shading" in the SQL connector distribution. I added exclusions so that the `flink-connector-base` classes are only packaged once. CI is running, let's see if it breaks anything.

>I'm also wondering whether we'd be able to add some test (bash based?) for the shaded artifacts, making sure this contract will hold with future changes

Would checking that there are no unshaded `org/apache/flink/connector/base/*` classes in the connectors would suffice? What do you think?

**Comment by dmvk** [comment]
> Would checking that there are no unshaded org/apache/flink/connector/base/* classes in the connectors would suffice? What do you think?

That plus checking that we only have a single shaded version per final artifact?

WDYT?

File: tools/ci/shade.sh [hunk 2]
```diff
@@ -173,3 +173,62 @@ check_shaded_artifacts_connector_elasticsearch() {
...
+	else
+		echo "OK"
+	fi
+}
+
+check_relocated() {
+    read foo
+	if [ $foo -ne 0 ]
+	then
+		echo "ERROR - CHECK FAILED: found $1 classses that where not relocated!"
```
Comment (zentol) [inline tools/ci/shade.sh:192]:
```suggestion
		echo "ERROR - CHECK FAILED: found $1 classes that where not relocated!"
```

File: flink-formats/flink-parquet/pom.xml [hunk 3]
```diff
@@ -308,6 +314,33 @@ under the License.
...
+						<configuration>
+							<artifactSet>
+								<includes>
+									<include>org.apache.flink:flink-connector-base</include>
+								</includes>
+							</artifactSet>
+							<relocations>
+								<relocation>
+									<pattern>org.apache.flink.connector.base</pattern>
+									<shadedPattern>org.apache.flink.connector.kinesis.shaded.org.apache.flink.connector.base</shadedPattern>
```
Comment (zentol) [inline flink-formats/flink-parquet/pom.xml:337]: pattern is incorrect

File: flink-formats/flink-orc/pom.xml [hunk 4]
```diff
@@ -204,6 +210,33 @@ under the License.
...
+						<configuration>
+							<artifactSet>
+								<includes>
+									<include>org.apache.flink:flink-connector-base</include>
+								</includes>
+							</artifactSet>
+							<relocations>
+								<relocation>
+									<pattern>org.apache.flink.connector.base</pattern>
+									<shadedPattern>org.apache.flink.connector.kinesis.shaded.org.apache.flink.connector.base</shadedPattern>
```
Comment (zentol) [inline flink-formats/flink-orc/pom.xml:233]: pattern is incorrect

File: flink-formats/flink-json/pom.xml [hunk 5]
```diff
@@ -77,6 +77,13 @@ under the License.
 
 		<!-- test dependencies -->
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-formats/flink-json/pom.xml:82]: why is this required? (same question for other formats)

**Review by zentol — CHANGES_REQUESTED** [review]
CI fails with a compile error.

**Comment by afedulov** [comment]
> CI fails with a compile error.

I am looking into this. The CI was green before the recent rebase on master. Something changed in the tests setup for Kinesis.

File: flink-connectors/flink-sql-connector-kafka/pom.xml [hunk 6]
```diff
@@ -40,6 +40,12 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-kafka</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
[the remark this answers is not in the document]
Comment (afedulov) [reply flink-connectors/flink-sql-connector-kafka/pom.xml:46]: @zentol without those exclusions, the package is included twice: once shaded, once non-shaded - see the comment from @dmvk https://github.com/apache/flink/pull/18905#issuecomment-1055498107.
Not sure why this happens.

File: flink-formats/flink-orc/pom.xml (hunk 4 above)
Comment (afedulov) [reply flink-formats/flink-orc/pom.xml]: fixed

File: flink-formats/flink-parquet/pom.xml (hunk 3 above)
Comment (afedulov) [reply flink-formats/flink-parquet/pom.xml]: fixed

File: flink-connectors/flink-sql-connector-kafka/pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (zentol) [reply flink-connectors/flink-sql-connector-kafka/pom.xml:46]: I guess what could happen is that some other dependency of connector-kafka may depend on connector-base. However I can't think of a scenario where we actually want that to result in it being bundled in downstream projects; as is that'd imply that if users bundle the kafka connector they'd also bundle connector-base twice.

so TL;DR: That you have to add this exclusion to me implies a larger issue.

File: tools/ci/shade.sh (hunk 2 above)
Comment (afedulov) [reply tools/ci/shade.sh]: fixed

File: flink-connectors/flink-sql-connector-kafka/pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (afedulov) [reply flink-connectors/flink-sql-connector-kafka/pom.xml]: I removed it from the maven-shade plugin instead.

File: flink-connectors/flink-sql-connector-kafka/pom.xml [hunk 7]
```diff
@@ -82,6 +88,10 @@ under the License.
 									<pattern>org.apache.kafka</pattern>
 									<shadedPattern>org.apache.flink.kafka.shaded.org.apache.kafka</shadedPattern>
 								</relocation>
+								<relocation>
+									<pattern>org.apache.flink.connector.base</pattern>
+									<shadedPattern>org.apache.flink.connector.kafka.sql.shaded.org.apache.flink.connector.base</shadedPattern>
```
[the remark this answers is not in the document]
Comment (afedulov) [reply flink-connectors/flink-sql-connector-kafka/pom.xml]: Removed.

File: flink-connectors/flink-connector-kinesis/pom.xml [hunk 8]
```diff
@@ -323,8 +323,10 @@ under the License.
 							<promoteTransitiveDependencies>true</promoteTransitiveDependencies>
 							<artifactSet combine.children="append">
 								<includes>
+									<include>org.apache.flink:flink-connector-aws-base:*</include>
```
Comment (zentol) [inline flink-connectors/flink-connector-kinesis/pom.xml:326]:
```suggestion
```

File: flink-connectors/flink-sql-connector-aws-kinesis-streams/pom.xml [hunk 9]
```diff
@@ -36,6 +36,12 @@
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-aws-kinesis-streams</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-aws-kinesis-streams/pom.xml:42]: should be unncessary because the only remaining transitive Flink dependency after shading is flink-table-common

File: flink-connectors/flink-sql-connector-elasticsearch6/pom.xml [hunk 10]
```diff
@@ -40,6 +40,12 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-elasticsearch6</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-elasticsearch6/pom.xml:46]: should be unncessary because there are no remaining transitive Flink dependencies after shading

File: flink-connectors/flink-sql-connector-elasticsearch7/pom.xml [hunk 11]
```diff
@@ -40,6 +40,12 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-elasticsearch7</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-elasticsearch7/pom.xml:46]: should be unncessary because there are no remaining transitive Flink dependencies after shading

File: flink-connectors/flink-sql-connector-aws-kinesis-firehose/pom.xml [hunk 12]
```diff
@@ -36,6 +36,12 @@
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-aws-kinesis-firehose</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-aws-kinesis-firehose/pom.xml:42]: should be unncessary because the only remaining transitive Flink dependency after shading is flink-table-common

File: flink-connectors/flink-sql-connector-kinesis/pom.xml [hunk 13]
```diff
@@ -40,6 +40,12 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-kinesis</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-kinesis/pom.xml:46]: should be unncessary because there are no remaining transitive Flink dependencies after shading

File: flink-connectors/flink-sql-connector-rabbitmq/pom.xml [hunk 14]
```diff
@@ -68,6 +68,10 @@ under the License.
 									<pattern>com.rabbitmq</pattern>
 									<shadedPattern>org.apache.flink.rabbitmq.shaded.com.rabbitmq</shadedPattern>
 								</relocation>
+								<relocation>
+									<pattern>org.apache.flink.connector.base</pattern>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-rabbitmq/pom.xml:72]: Why is this done in the sql connector, and not the connector itself?

File: flink-connectors/flink-sql-connector-pulsar/pom.xml [hunk 15]
```diff
@@ -40,6 +40,12 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-pulsar</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-pulsar/pom.xml:46]: should be unncessary because there are no remaining transitive Flink dependencies after shading

**Comment by MartijnVisser** [comment]
@flinkbot run azure

File: flink-end-to-end-tests/flink-end-to-end-tests-aws-kinesis-firehose/pom.xml [hunk 16]
```diff
@@ -51,6 +51,14 @@
 			<scope>test</scope>
 			<type>test-jar</type>
 		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-aws-base</artifactId>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-aws-kinesis-firehose/pom.xml:57]: its a bit weird that this is now (supposedly) required. Irrespective of the changes made, aws-base wasn't on the classpath in the e2e test because so far it only used test-jars which don't bring in transitive dependencies.

File: flink-end-to-end-tests/flink-end-to-end-tests-aws-kinesis-streams/pom.xml [hunk 17]
```diff
@@ -65,6 +65,13 @@
 			<artifactId>jackson-databind</artifactId>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-aws-base</artifactId>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-aws-kinesis-streams/pom.xml:71]: same question as for firehose. Since they don't use the connector directly (but add the sql-connector jar to the jobs classpath) it shouldn't be needed.

File: flink-end-to-end-tests/flink-streaming-kinesis-test/pom.xml [hunk 18]
```diff
@@ -88,6 +88,13 @@ under the License.
 			<type>test-jar</type>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-aws-base</artifactId>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-streaming-kinesis-test/pom.xml:93]: same question as for firehose

File: flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/pom.xml [hunk 19]
```diff
@@ -156,6 +156,13 @@ under the License.
 			<version>${project.version}</version>
 			<type>test-jar</type>
 		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-common-kafka/pom.xml:162]: same question as for firehose

File: flink-formats/flink-csv/pom.xml [hunk 20]
```diff
@@ -76,6 +76,12 @@ under the License.
 		</dependency>
 
 		<!-- test dependencies -->
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-formats/flink-csv/pom.xml:81]: What needs this? There is no import in this module to any of the classes, and everything else that might need it (be it a connector or format) should bring their own version.

File: flink-table/flink-sql-client/pom.xml [hunk 21]
```diff
@@ -73,6 +73,13 @@ under the License.
 
 		<!-- test dependencies -->
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-table/flink-sql-client/pom.xml:78]: same question as for the csv-format

File: flink-table/flink-table-planner/pom.xml [hunk 22]
```diff
@@ -198,6 +198,13 @@ under the License.
 			<scope>test</scope>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-table/flink-table-planner/pom.xml:203]: same question as for the csv-format

File: flink-yarn-tests/pom.xml [hunk 23]
```diff
@@ -41,6 +41,13 @@ under the License.
 		
 		<!-- test dependencies -->
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-yarn-tests/pom.xml:46]: same question as for the csv-format

File: flink-formats/flink-avro/pom.xml [hunk 24]
```diff
@@ -92,6 +92,13 @@ under the License.
 			<scope>test</scope>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-formats/flink-avro/pom.xml:97]: same question as for the csv-format

File: flink-examples/flink-examples-table/pom.xml [hunk 25]
```diff
@@ -93,6 +93,13 @@ under the License.
 			<version>${project.version}</version>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-examples/flink-examples-table/pom.xml:99]: same question as for the csv-format

File: flink-examples/flink-examples-streaming/pom.xml [hunk 26]
```diff
@@ -80,6 +80,13 @@ under the License.
 
 		<!-- test dependencies -->
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-base</artifactId>
```
Comment (zentol) [inline flink-examples/flink-examples-streaming/pom.xml:85]: same question as for the csv-format

**Review by zentol — CHANGES_REQUESTED** [review]
I think there are still quite a few instances of blanket changes being applied despite not really being necessary.

**Review by zentol — APPROVED** [review]
